utils/docker/Dockerfile.centos.7

```diff
@@ -160,6 +159,8 @@ COPY --chown=daos_server:daos_server site_scons site_scons
 COPY --chown=daos_server:daos_server utils utils
 COPY --chown=daos_server:daos_server src src
 
+# select compiler to use
+ARG COMPILER=gcc
```

Comment:
This was moved up higher to be in parity with the compiler selection code in the Leap 15 Dockerfile.  Arguably, all Dockerfiles should have this compiler selection code so that one could choose _icc_ on any platform.

While I agree that `ARG`s should be placed as close to their first use as possible for cache efficiency, if the `ARG` is not even used in a given Dockerfile it doesn't really matter where it goes, so keeping it in the same place in all files (for easy side-by-side viewing/comparison) makes the most sense.